docs/samples/Microsoft.ML.Samples/Dynamic/LightGBM/LightGBMMulticlassClassification.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.SamplesUtils;
+
+namespace Microsoft.ML.Samples.Dynamic.LightGBM
+{
+    class LightGbmMulticlassClassification
+    {
+        public static void LightGbmMulticlassClassificationExample()
+        {
+            // Create a general context for ML.NET operations. It can be used for exception tracking and logging,
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Create in-memory examples as C# native class.
+            var examples = DatasetUtils.GenerateRandomMulticlassClassificationExamples(1000);
+
+            // Convert native C# class to IDataView, a consumble format to ML.NET functions.
+            var dataView = mlContext.Data.ReadFromEnumerable(examples);
+
+            // Create a pipeline. 
+            //  - Convert the string labels into key types.
+            //  - Apply LightGbm multiclass trainer
+            var pipeline = mlContext.Transforms.Conversion.MapValueToKey("LabelIndex", "Label")
+                        .Append(mlContext.MulticlassClassification.Trainers.LightGbm(labelColumn: "LabelIndex"))
```

Comment:
>LightGbm [](start = 76, length = 8)

from gitter:
`Why does LightGbmMulticlassTrainer constructor when called with Options callMakeBoolScalarLabel? That throws for me if I try a multi class problem with options. If I call the other constructor it's workign fine`

You probably hit same problem if you try to use options.

Reply:
OK, I fixed that bug too.

---
In reply to: [255322542](https://github.com/dotnet/machinelearning/pull/2476#discussion_r255322542) [](ancestors = 255322542)